### PR TITLE
[html-report] Display unreachable code as Any 

### PR DESCRIPTION
**Pull request by ilinum**

### CODE REVIEW

**Comment by gvanrossum** [comment]
However you need to figure out why the Windows tests are failing. Probably something with path separators (it always is, except when it's LF vs. CRLF :-).

**Comment by ilinum** [comment]
As it turns out, the path to the CSS file on windows was wrong. It was trying to use the native filepath. However, on Windows the `\` would get converted to `%5C`, so path to CSS file would look like `..%5Cmypy-html.css`, which is wrong.

What this means is that css was not working on windows and lines in html report did not have colors. This is fixed now.

**Comment by ilinum** [comment]
I realized that this PR does not address reporting of dead code in `--any-expr-report`.
I don't think it's ready to be merged until this is done.

**Comment by ilinum** [comment]
This PR is ready for review!

Now `--any-expr-report` counts each line of dead code as `Any`.

File: mypy/stats.py [hunk 1]
```diff
@@ -152,12 +161,17 @@ def visit_unary_expr(self, o: UnaryExpr) -> None:
...
+            self.line = node.line
+            self.type(typ)
+
+    def type(self, t: Optional[Type]) -> None:
+        if not t:
+            # if an expression does not have a type, it is often due to dead code
+            # we should not keep track of the number of these we encounter because there can be
+            # an unanalyzed value on a line with other analyzed expressions
+            self.record_line(self.line, TYPE_UNANALYZED)
+            return
```
Comment (ddfisher) [inline mypy/stats.py]: Nit: probably better to change the `if isinstance(t, Instance)` to an `elif` below instead of this early return.

Comment (ilinum) [inline mypy/stats.py]: I don't think that would work - there are two separate if-statements and I think it's easier to just address it in the beginning and do an early return.

File: mypy/stats.py [hunk 2]
```diff
@@ -152,12 +161,17 @@ def visit_unary_expr(self, o: UnaryExpr) -> None:
...
-    def type(self, t: Type) -> None:
-        if isinstance(t, AnyType):
+            self.line = node.line
+            self.type(typ)
+
+    def type(self, t: Optional[Type]) -> None:
+        if not t:
+            # if an expression does not have a type, it is often due to dead code
+            # we should not keep track of the number of these we encounter because there can be
+            # an unanalyzed value on a line with other analyzed expressions
```
Comment (ilinum) [inline mypy/stats.py]: ok!

File: mypy/stats.py [hunk 3]
```diff
@@ -79,6 +81,13 @@ def visit_func_def(self, o: FuncDef) -> None:
                 self.record_line(self.line, TYPE_ANY)
             super().visit_func_def(o)
 
+    def visit_class_def(self, o: ClassDef) -> None:
+        # this method is overridden because we don't want to visit base_type_exprs
+        # base_type_exprs are expressions without a type, which causes them to appear as Any
```
Comment (ilinum) [inline mypy/stats.py]: Yep, hopefully, should be better now.

File: mypy/stats.py [hunk 4]
```diff
@@ -12,17 +12,19 @@
...
-    MemberExpr, OpExpr, ComparisonExpr, IndexExpr, UnaryExpr, YieldFromExpr, RefExpr
+    MemberExpr, OpExpr, ComparisonExpr, IndexExpr, UnaryExpr, YieldFromExpr, RefExpr, ClassDef
 )
 
 
 TYPE_EMPTY = 0
-TYPE_PRECISE = 1
-TYPE_IMPRECISE = 2
-TYPE_ANY = 3
+TYPE_UNANALYZED = 1  # type of unchecked code (different from any)
```
Comment (ilinum) [inline mypy/stats.py]: Updated.

**Pull request merged**